### ISSUE TITLE
Adding different Box layouts example

### ISSUE THREAD

**Issue by arthmis** [linebender/druid#1673]
So there is a layout example currently. I could add `AspectRatioBox` to that layout example but I feel like it would be lost amongst the other elements. Could I add an example that shows off the different types of "sized boxes". Eventually there will be, `AspectRatio`, `ConstrainedBox`, UnconstrainedBox`, `FractionallySizedBox` and a few more I believe. I feel like all of these and `SizedBox`, should be within one example that demonstrates them all.

**Comment by cmyr** [comment]
I would ultimately like to reduce the total number of examples, but I can see how your idea could be helpful. What I think might be helpful is eventually splitting out the 'layout' widgets from other widgets (a `widgets::layout` module?) And then having good documentation there that describes the different layout widgets and what you might want to use them for.

**Comment by arthmis** [comment]
Do you count things like `FractionallySizedBox` and `AspectRatioBox` as layout? If so then, maybe there could be something like https://flutter.dev/docs/development/ui/layout/constraints but for druid. Maybe it could be a tour of layout and pitfalls to look out for. So instead of thinking of how to understand constraints, it can go into how to use the widgets to affect layout. The tour of layout and how to use the different widgets could maybe go in the book. The actual docs could link to this tour. I'm not too sure. 

Maybe there could also be a place to write about constraints, but this should probably go in the Druid book. The layout model shouldn't change right? So this kind of documentation wouldn't have to go through a massive rewrite, I hope. 

I agree it will eventually make sense to have a layout module as more of these box types are added. 

Since you want to reduce examples? Do you want to move some of the examples to the docs for each widget?

**Comment by cmyr** [comment]
Yes, I think a layout widget is any widget that modifies the constraints passed to its child, or otherwise influences the child's layout; and generally it also does *not* do other things, like handle events  or lifecycle (except to propagate changes from children, as needed).
